### PR TITLE
This is the pull request for all of lesson 2

### PR DESCRIPTION
**Pull request by hood821**
Added all the assignments for lesson 2.

### CODE REVIEW

File: students/TressaHood/lesson02/fizz_buzz.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#lesson 2, assignment fizz buzz
+
+#define the main function
+def main():
+
+	#print 1 - 100 inclusive
+	for i in range(1,101):
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/fizz_buzz.py]: You'll want 4 space indentions for Python.  Recommend removing the extra blank lines within your function as well.

File: students/TressaHood/lesson02/fizz_buzz.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		#if multiple of 5 print buzz
+		elif( i % 5 == 0):
+			print("Buzz")
+
+		else:
+			print(i)
+
+
+#call the main function
+main()
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/fizz_buzz.py]: Works well!

File: students/TressaHood/lesson02/grid_printer.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#part 2, create the sections with a function
+#define the print grid function
+def print_grid_part2(x):
+	size = x//2
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/grid_printer.py]: 4 spaces for indents

File: students/TressaHood/lesson02/grid_printer.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+#define the main function
+def main():
+
+	#part 1
+	#create my sections manually
+	first = ("+ " + "- "*4)*2 + "+"
+	second = ("\n" + ("|" + "\t " + " " + "\t " + " ")*2 + "  |")*4
+	print("This is part one:")
+	print(first, second, "\n" + first, second, "\n" + first)
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/grid_printer.py]: This one doesn't quite seem to come our right.

File: students/TressaHood/lesson02/grid_printer.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+	#now do part 3
+	print("This is part three:")
+	print_grid_part3(3,4)
+	print_grid_part3(5,3)
+
+
+
+#call the main function
+main()
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/grid_printer.py]: Overall works well.

File: students/TressaHood/lesson02/series.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+#lesson 2, assignment fibonacci series
+
+def fibonacci(n):
+	"""
+	Return the nth value of the fibonaaci sequence using recursion. The function
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/series.py]: Indent 4 spaces.

File: students/TressaHood/lesson02/series.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+	calls it self until it has finished (n=0). This sequence starts with 0 and 1.
+
+	:param n: The number within in the fibonacci sequence you wish to extract.
+	"""
+	if(n<=1):
+		return n
+	else:
+		return fibonacci(n-2) + fibonacci(n-1)
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/series.py]: Works well

File: students/TressaHood/lesson02/series.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+	calls it self until it has finished (n=0). This sequence starts with 2 and 1.
+
+	:param n: The number within in the lucas sequence you wish to extract.
+	"""
+	if(n==0):
+		return 2
+	elif(n==1):
+		return 1
+	else:
+		return lucas(n-2) + lucas(n-1)
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/series.py]: Works well as well.

File: students/TressaHood/lesson02/series.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+	assert sum_series(5) == 5
+
+	#test the lucas version of sum_series
+	assert sum_series(5,2,1) == 11
+
+	#test a random sequence
+	assert sum_series(5, 3, 5) == 34
+
+	#finish testing
+	print("Test completed successfully :)")
```
Comment (charlesrobison) [inline students/TressaHood/lesson02/series.py]: Good assert statements.

File: students/TressaHood/lesson02/fizz_buzz.py (hunk 1 above)
Comment (hood821) [reply students/TressaHood/lesson02/fizz_buzz.py]: It's weird because in my tab preferences it says 4? not sure why this is.

**Comment by hood821** [comment]
I've also added all of lesson 3, I am not sure how to do another pull request? Or if I don't?

**Pull request merged**